### DISCUSSION TITLE
Support for integrating 3rd party PII sensitive data protection with Guardrails flow

### DISCUSSION THREAD

**Discussion by avirgvd** [NVIDIA-NeMo/Guardrails#294] [Q&A] [answered]
Guardrails have built-in library that uses Presidio for detecting and masking sensitive data. How can I use a 3rd party alternative to Presidio for sensitive data protection with Guardrails?
I am looking for any integration support in Guardrails that allows me to plugin any 3rd party PII sensitive data protection such as PrivateAI.

**Comment by drazvan** [comment, marked as answer]
Yes. You can connect any 3rd party. If you follow the implementation pattern for Presidio, you can do anything you want in the `actions.py` file:
https://github.com/NVIDIA/NeMo-Guardrails/tree/develop/nemoguardrails/library/sensitive_data_detection

And another good example, which actually uses a 3rd party API is this: https://github.com/NVIDIA/NeMo-Guardrails/tree/develop/nemoguardrails/library/activefence.

**Reply by avirgvd** [reply]
Thank you!
